type-definitions/immutable.js.flow

```diff
@@ -561,9 +561,78 @@ declare module 'immutable' {
     flatten(shallow?: boolean): /*this*/Map<any,any>;
   }
 
-  // OrderedMaps have nothing that Maps do not have. We do not need to override constructor & other statics
-  declare class OrderedMap<K,V> extends Map<K,V> {
+  declare class OrderedMap<K,V> extends KeyedCollection<K,V> {
```

Comment:
Turns out I'm getting an issue overriding `update` by extending Map.

![image](https://cloud.githubusercontent.com/assets/831651/21411981/9239b762-c7ba-11e6-9076-0a0aa195b63a.png)

I'll leave it as extending `KeyedCollection` for the purposes of this PR. Having `OrderedSet` extend `Set` worked without issues (I didn't realize I was already doing that), and I'll update the PR to remove extraneous OrderedSet definitions.